### ISSUE TITLE
`:nodoc:` on enum member does not also hide the generated question method

### ISSUE THREAD

**Issue by Blacksmoke16** [crystal-lang/crystal#11333]
If you define an enum with a `:nodoc:` member, like:

```cr
enum Test
  FOO

  # :nodoc:
  BAR
end
```

The member itself will be hidden, but the autogenerated `bar?` method ill not:

![image](https://user-images.githubusercontent.com/12136995/137644397-4e72197c-a79b-4d8b-ba74-22f9704f1e22.png)

I think it would make sense to also hide the method.

**Comment by asterite** [comment]
Pretty hard, the regular compiler doesn't know about doc comments.

**Comment by straight-shoota** [comment]
Thanks to @nobodywasishere for pushing #15375. But I'm not quite sure that's the correct solution for this.
In principle, duplicating documentation is quite poor, especially when it's not explicitly declared for the user. They'll have to read both documentations without realizing that they're identical and one or the other would suffice.

Instead, it should be better to refer to the primary documentation instead of copying the contents.

This particular issue didn't even ask to duplicate the documentation of the enum member. It's just about making derived predicate methods `nodoc`. It should be possible to do that without copying the entire doc comment.
However, it might be a good idea to generate a generic doc comment for enum methods which describe the function and refer to the member.
Something like this:
```cr
# Returns `true` if this enum value {{ flag_type? ? "contains" : "equals" }} `{{ member }}`.
```

**Comment by nobodywasishere** [comment]
Fair enough, I'll update it accordingly. Adding that documentation also sounds like a good idea. I can check if the enum member starts with `:nodoc:` and append it to the helper method if necessary.
